tests/mock_requests.py

```diff
@@ -0,0 +1,89 @@
+"""
+Mock the `requests.get` function, and handle collecting data to do so.
+
+Three modes are available, and controlled via the `PUBS_TESTS_MODE` environment
+variable. To modify the variable, under linux or macos, do one of:
+$ export PUBS_TESTS_MODE=MOCK
+$ export PUBS_TESTS_MODE=COLLECT
+$ export PUBS_TESTS_MODE=ONLINE
+
+The MOCK mode is the default one, active even if PUBS_TESTS_MODE has not been
+set. It use prefetched data to run pubs units tests relying on the `requests.get`
+function without the need of an internet connection (it is also much faster).
+The prefected data is save in the `test_apis_data.pickle` file.
+
+The COLLECT mode does real GET requests, and update the `test_apis_data.pickle`
+file. It is needed if you add or modify the test relying on `requests.get`.
+
+The ONLINE mode bypasses all this and use the original `requests.get` without
+accessing or updating the `test_apis_data.pickle` data. It might be useful when
+running tests on Travis for instance.
+"""
+
+
+import os
+try:
+    import cPickle as pickle
```

Comment:
Wait wait. `cPickle` is not a dependency. It's the name of the the builtin C implementation of the pickle module in Python 2.7. Python 3 renamed cPickle into pickle and (only?) has the C-implementation. This is just meant to use the faster version for Python 2.7. Probably not that significant a speed boost, but I thought it would mean using more similar implementation between Python 2 and 3, but on reading the doc, Python 2.7's pickle and cPickle actually produce identical info (https://docs.python.org/2/library/pickle.html#module-cPickle).

Compatibility between Python 2 and 3 is ensured by calling `dump` with `protocol=2`, the latest protocol supported by Python 2.7.

About using json versus pickle, well, pickle generates smaller files, that are read faster. Current tests generate 43K of data, pickled. I used pickle to avoid the repository size getting large as we modify tests.

Reply:
>About using json versus pickle, well, pickle generates smaller files, that are read faster. 

Are you sure about that? I don't see any reason why it would be the case since the data is mostly text. Also, it is easy to compress so should not affect the repository size in a sensible way. And if we kind of sort the data before dumping, the json file should also be versioned easily so will impact less the repository size than a binary file that is modified each time.

There is also a serious security concern related to using pickle which is that it would be equivalent to potentially distributing code that is not directly readable. Even if we trust ourselves, we would have to re-generate the pickle file ourselves each time somebody contributes a commit that changes it to be able to trust its content. We could organize around that but it seems that there are many concerns here for a very speculative gain.